File: backend_nestjs/src/animals/entities/animal.entity.ts
import { Corral } from 'src/corrals/entities/corral.entity';
import { Restriction } from 'src/restriction/entities/restriction.entity';
import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  ManyToOne,
  OneToMany
} from 'typeorm';

@Entity()
export class Animal {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column()
  age: number;

  @ManyToOne(() => Corral, (corral) => corral.animals)
  corral: Corral;

  @OneToMany(() => Restriction, (restriction) => restriction.animal)
  restrictions: Restriction[];
}

File: BackendSpringboot/src/main/java/RIwi/BackendSpringboot/model/entities/AnimalEntity.java
package RIwi.BackendSpringboot.model.entities;
import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;


@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Entity(name = "Animal")
public class AnimalEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Name cannot be null")
    @NotEmpty(message = "Name cannot be empty")
    private String name;

    @Min(value = 0, message = "Age cannot be negative")
    private int age;

    @NotNull(message = "'dangerous' field cannot be null")
    private boolean dangerous;

    @ManyToMany
    @JoinTable(
            name = "animal_restriction",
            joinColumns = @JoinColumn(name = "animal_id"),
            inverseJoinColumns = @JoinColumn(name = "restricted_animal_id")
    )
    private List<AnimalEntity> incompatibleAnimals = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "corral_id")
    private CorralEntity corral;

}
